packages/ra-core/src/util/hooks.ts

```diff
@@ -62,3 +62,11 @@ export function useTimeout(ms = 0) {
 
     return ready;
 }
+
+export function useDocumentTitle(title: string) {
```

Comment:
I'm fine with having this hook in ra-core, but I think it deserves its own file and a jsDoc block with example usage. You can keep it in util/.